numpyro/distributions/mixtures.py

```diff
@@ -389,9 +386,21 @@ def component_sample(self, key, sample_shape=()):
         return jnp.stack(samples, axis=self.mixture_dim)
 
     def component_log_probs(self, value):
-        component_log_probs = jnp.stack(
-            [d.log_prob(value) for d in self.component_distributions], axis=-1
-        )
+        if self._validate_args:
+            mask = jnp.stack(
+                [d.support(value) for d in self.component_distributions], axis=-1
+            )
+            component_log_probs = jnp.where(
+                mask,
+                jnp.stack(
+                    [d.log_prob(value) for d in self.component_distributions], axis=-1
```

Comment:
It is a little bit cheaper to avoid the duplication of the masking logic
```
component_log_probs = []
for d in self.component_distributions:
  log_prob = d.log_prob(value)
  if (self._support is not None) and (not d._validate_args):
    mask = d.support(value)
    log_prob = jnp.where(mask, log_prob, -jnp.inf)
  component_log_probs.append(log_prob)
...
```

Reply:
This makes more sense.

Reply:
> ```python
> if (self._support is not None) and (not d._validate_args):
> ```
I think `not` is not required with `d._validate_args`.